solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -763,8 +765,53 @@ contract LiquidityProtection is ILiquidityProtection, Utils, Owned, ReentrancyGu
         validAddress(newProvider)
         returns (uint256)
     {
+        return transferPosition(msg.sender, id, newProvider);
+    }
+
+    /**
+     * @dev transfers a position to a new provider and optionally notifies another contract
+     *
+     * @param id position id
+     * @param newProvider the new provider
+     * @param target the contract to notify
+     * @param data the data to call the contract with
+     *
+     * @return new position id
+     */
+    function transferPositionAndCall(
+        uint256 id,
+        address newProvider,
+        address target,
+        bytes memory data
+    ) external protected validAddress(newProvider) returns (uint256) {
+        uint256 newId = transferPosition(msg.sender, id, newProvider);
+
+        // make sure that we're not trying to call into the zero address or a fallback function
+        require(target != address(0) && data.length >= FUNC_SELECTOR_LENGTH, "ERR_INVALID_CALL_DATA");
```

Comment:
any reason not to call `validAddress` on the target instead?

Reply:
@barakman mentioned it as well, so I'll just fix it 😎